### PR TITLE
coap-gateway: update pending commands via eventbus

### LINKED ISSUE #378: Gateways receive pending commands from EventBus
### What is the purpose of the feature?
plgd Gateways are receiving events about pending commands from the Resource Directory, instead of being subscribed directly to the EventBus. The goal is to not use this middleware forwarding EventBus events through the gRPC stream but directly subscribe to the EventBus.

### What specific challenge does this feature solve?
Reduces unnecessary dependency introducing additional complexity during scaling.

### What alternatives did I consider?
None

### PR DESCRIPTION
**Pull request by jkralik**

### CODE REVIEW

File: bundle/run.sh [hunk 1]
```diff
@@ -162,6 +162,165 @@ certificate-generator --cmd.generateIdentityCertificate=$COAP_GATEWAY_CLOUD_ID -
 echo "generating NGINX cert"
 certificate-generator --cmd.generateCertificate --outCert=$EXTERNAL_CERT_DIR_PATH/$DIAL_FILE_CERT_NAME --outKey=$EXTERNAL_CERT_DIR_PATH/$DIAL_FILE_CERT_KEY_NAME --cert.subject.cn="localhost" --cert.san.domain="localhost" --cert.san.ip="0.0.0.0" --cert.san.ip="127.0.0.1" $fqdnSAN --signerCert=$CA_POOL_CERT_PATH --signerKey=$CA_POOL_CERT_KEY_PATH
 
+# copy ceritficates to paths
```
Comment (ondrejtomcik) [inline bundle/run.sh]: What has this to do with `coap-gateway: update pending commands via eventbus`? :)

File: coap-gateway/service/client.go [hunk 2]
```diff
@@ -142,23 +146,29 @@ func (client *Client) cancelResourceSubscription(token string, wantWait bool) (b
...
 }
 
+func copyString(a string) string {
+	return (a + " ")[:len(a)]
+}
+
 func (client *Client) observeResource(ctx context.Context, resourceID *commands.ResourceId, observable, allowDuplicit bool) (err error) {
 	log.Debugf("observation of resource %v requested", resourceID)
 	instanceID := getInstanceID(resourceID.GetHref())
+	deviceID := copyString(resourceID.GetDeviceId())
```
Comment (ondrejtomcik) [inline coap-gateway/service/client.go]: Why do you need to make a copy?

File: coap-gateway/service/service.go [hunk 3]
```diff
@@ -493,7 +386,7 @@ func (server *Service) setupCoapServer() {
 	}))
 
 	opts := make([]tcp.ServerOption, 0, 8)
-	opts = append(opts, tcp.WithKeepAlive(1, server.config.APIs.COAP.KeepAlive.Timeout, server.KeepaliveOnInactivity))
+	opts = append(opts, tcp.WithKeepAlive(1, server.config.APIs.COAP.KeepAlive.Timeout/2, server.KeepaliveOnInactivity))
```
Comment (ondrejtomcik) [inline coap-gateway/service/service.go]: Why /2 ?

**Review by ondrejtomcik — APPROVED** [review]

File: coap-gateway/service/service.go (hunk 3 above)
Comment (ondrejtomcik) [reply coap-gateway/service/service.go]: Move to go-coap

File: coap-gateway/service/client.go (hunk 2 above)
Comment (jkralik) [reply coap-gateway/service/client.go]: removed

**Pull request merged**